source/common/router/rds_impl.cc

```diff
@@ -38,6 +38,13 @@ StaticRouteConfigProviderImpl::StaticRouteConfigProviderImpl(
     Upstream::ClusterManager& cm)
     : config_(new ConfigImpl(config, runtime, cm, true)) {}
 
+std::string RdsRouteConfigProviderImpl::rdsStatsName(const std::string& stat_prefix,
+                                                     const std::string& route_config_name) {
+  std::string stats_name = fmt::format("{}rds.{}.", stat_prefix, route_config_name);
+  std::replace(stats_name.begin(), stats_name.end(), ':', '_');
```

Comment:
Do you have a test case for this? cc @hennna please update this site also in your other PR if we move this into scope.

Reply:
I used all the stats expectations as a test that the name is correct. But I will make an explicit test.

Reply:
I've updated #1806. Let me know if I'm missing anything else.

Reply:
will remove this in favor of @hennna's change.